Find a functionally equivalent code:com.android.smspush.WapPushManager.WapPushManDBHelper.queryLastApp:COMMENT
Method Modifier: protected   
Comment:/**
 * Query the latest receiver application info with supplied application ID and
 * content type.
 * @param app_id    application ID to look up
 * @param content_type    content type to look up
 */

@@ -1,8 +1,9 @@
 {
-    String sql = "select install_order, package_name, class_name, " + " app_type, need_signature, further_processing" + " from " + APPID_TABLE_NAME + " where x_wap_application=\'" + app_id + "\'" + " and content_type=\'" + content_type + "\'" + " order by install_order desc";
-    if (DEBUG_SQL)
-        Log.v(LOG_TAG, "sql: " + sql);
-    Cursor cur = db.rawQuery(sql, null);
+    if (LOCAL_LOGV)
+        Log.v(LOG_TAG, "queryLastApp app_id: " + app_id + " content_type: " + content_type);
+    Cursor cur = db.query(APPID_TABLE_NAME, new String[] { "install_order", "package_name", "class_name", "app_type", "need_signature", "further_processing" }, "x_wap_application=? and content_type=?", new String[] { app_id, content_type }, null, /* groupBy */
+    null, /* having */
+    "install_order desc");
     queryData ret = null;
     if (cur.moveToNext()) {
         ret = new queryData();

